### PR TITLE
[Enhancement] Collect file write history

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Problem Summary(Required) ：

This PR collects file write the information of BE and put them into a history buffer, for easy debugging file IO related problems.

```
mysql> admin execute on 10004 '
    '> for (f in ExecEnv.get_file_write_history()) {
    '>     if (!f.path.contains(".testfile")) {
    '>         System.print("file: %(f.path) size: %(f.size) open_time: %(f.open_time)")
    '>     }
    '> }
    '> ';
Query OK, 0 rows affected (0.01 sec)
file: /home/decster/projects/starrocks/localrun/be/storage/.read_write_test_file size: 4096 open_time: 1682253498
file: /home/decster/projects/starrocks/localrun/be/storage/data/0/10009/109317232/0200000000000001a148ac7b1f5aafe038674dd9ff13ceb9_0.dat size: 479 open_time: 1682253543
file: /home/decster/projects/starrocks/localrun/be/storage/data/0/10009/109317232/0200000000000002a148ac7b1f5aafe038674dd9ff13ceb9_0.dat size: 479 open_time: 1682253544
file: /home/decster/projects/starrocks/localrun/be/storage/data/0/10009/109317232/0200000000000003a148ac7b1f5aafe038674dd9ff13ceb9_0.dat size: 479 open_time: 1682253558
file: /home/decster/projects/starrocks/localrun/be/storage/data/18/10054/430336460/0200000000000015a148ac7b1f5aafe038674dd9ff13ceb9_0.dat size: 1892 open_time: 1682253801
file: /home/decster/projects/starrocks/localrun/be/storage/data/20/10058/430336460/0200000000000017a148ac7b1f5aafe038674dd9ff13ceb9_0.dat size: 1928 open_time: 1682253801

```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by trueeyu — APPROVED** [review]

File: be/src/fs/fs.h [hunk 1]
```diff
@@ -104,6 +104,13 @@ struct DirEntry {
     std::optional<bool> is_dir;
 };
 
+struct FileWriteStat {
+    int64_t open_time{0};
+    int64_t close_time{0};
+    int64_t size{0};
+    std::string path;
```
Comment (ABingHuang) [inline be/src/fs/fs.h:111]: is it absolute path or relative path?

Comment (decster) [reply be/src/fs/fs.h:111]: absolute path, for all type of fs

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**